SideBar.py

```diff
@@ -72,25 +89,52 @@ def run(self, paths):
 	def on_done(self, destination):
 		base, _ = os.path.split(self.source)
 		destination = os.path.join(base, destination)
-		threading.Thread(target=self.copy,
-			args=(self.source, destination)).start()
+		threading.Thread(target=self.copy, args=(self.source, destination)).start()
 
 	def copy(self, source, destination):
-		print(source, destination)
-		if self.view:
-			self.view.set_status('ZZZ', 'copying "{}" to "{}"'.format(
-				source, destination))
-		else:
-			self.window.status_message('copying "{}" to "{}"'.format(
-				source, destination))
+		self.window.status_message('Copying "{}" to "{}"'.format(source, destination))
+
+		self.make_dirs_for(destination)
 
 		if os.path.isdir(source):
 			shutil.copytree(source, destination)
 		else:
 			shutil.copy2(source, destination)
 
-		if self.view:
-			self.view.erase_status('ZZZ')
-
 	def description(self):
 		return 'Duplicate File…'
+
+
+class SideBarMoveCommand(SideBarCommand):
+
+	def run(self, paths):
+		self.source = self.get_path(paths)
+
+		input_panel = self.window.show_input_panel(
+			'Move to:', self.source, self.on_done, None, None)
+
+		base, leaf = os.path.split(self.source)
+		name, ext = os.path.splitext(leaf)
+
+		input_panel.sel().clear()
+		input_panel.sel().add(sublime.Region(len(base) + 1, len(self.source) - len(ext)))
+
+	def on_done(self, destination):
+		threading.Thread(target=self.move, args=(self.source, destination)).start()
+
+	def move(self, source, destination):
+		self.window.status_message('Moving "{}" to "{}"'.format(source, destination))
+
+		self.make_dirs_for(destination)
+
+		try:
+			shutil.move(source, destination)
+		except OSError as error:
+			self.window.status_message('Error moving "{src}" to "{dst}": {error}'.format(
```

Comment:
Well, I still like "my" style, but the counter argument is that space in the status bar is scarce, so I think your style makes more sense in this case.